src/viam/ur/module/ur_arm.cpp

```diff
@@ -185,9 +203,35 @@ std::vector<std::shared_ptr<ModelRegistration>> URArm::create_model_registration
 }
 
 URArm::URArm(Model model, const Dependencies& deps, const ResourceConfig& cfg) : Arm(cfg.name()), model_(std::move(model)) {
-    VIAM_SDK_LOG(info) << "URArm constructor start (model: " << model_.to_string() << ")";
+    VIAM_SDK_LOG(info) << "URArm constructor called (model: " << model_.to_string() << ")";
+    startup_(deps, cfg);
+}
+
+void URArm::startup_(const Dependencies&, const ResourceConfig& cfg) {
+    if (current_state_) {
+        throw std::logic_error("URArm::startup_ was called for an already running instance");
+    }
```

Comment:
[q] When are we expecting to startup an already running instance? Or is this a guard against this function being called from another one of the Arm's APIs.

I would expect this to only be called in the constructor for this driver.

Reply:
This gets called again if you reconfigure, in order to start up with the newly provided configuration. The check here ensures that `startup_` is never called with an active configuration already in place.